### PR TITLE
fix(column): avoid exceeding configured 'signcolumn' width

### LINKED ISSUE #26685: Diagnostic signs do not respect `signcolumn=yes:N` setting when `N > 1`
### Problem

Don't know if I'm doing something wrong, but it seems that in the latest nightly diagnostic signs do not respect the `signcolumn` config. 

Here's a screenshot of the behaviour: 

image

---
As visible in the screenshot `signcolumn` is set to `yes:2`. If I'm not mistaken, I should be able to see the sign provided by `Gitsigns` and one diagnostic sign. 

If more diagnostics are present the sign column might even be expanded further: 

image

---

When no diagnostics are present, sign column shrinks to the configured size: 

image

---

Weirdly enough, this does not happen when `signcolumn=yes:1`

### Steps to reproduce

Setup the diagnostic signs:

```lua
vim.cmd("set signcolumn=yes:2")

-- setup client, not really relevant
local client_id = vim.lsp.start({
	name = "rust-analyzer",
	cmd = { "rust-analyzer" },
	root_dir = vim.fs.dirname(vim.fs.find({ "Cargo.toml" }, { upward = true })[1]),
})

if client_id then
	vim.lsp.buf_attach_client(0, client_id)
end

-- setup signs, relevant
local signs = { Error = " ", Warn = " ", Hint = " ", Info = " " }
vim.diagnostic.config({
	virtual_text = true,
	signs = {
		enable = true,
		text = {
			["ERROR"] = signs.Error,
			["WARN"] = signs.Warn,
			["HINT"] = signs.Hint,
			["INFO"] = signs.Info,
		},
		texthl = {
			["ERROR"] = "DiagnosticDefault",
			["WARN"] = "DiagnosticDefault",
			["HINT"] = "DiagnosticDefault",
			["INFO"] = "DiagnosticDefault",
		},
		numhl = {
			["ERROR"] = "DiagnosticDefault",
			["WARN"] = "DiagnosticDefault",
			["HINT"] = "DiagnosticDefault",
			["INFO"] = "DiagnosticDefault",
		},
		severity_sort = true,
	},
})

vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(vim.lsp.diagnostic.on_publish_diagnostics, {
	virtual_text = true,
	signs = true,
	update_in_insert = false,
	underline = true,
})
```

Then try to mess up code to get multiple diagnostics on the same line. The sign column expands to make space for multiple diagnostics (not wanted). 

### Expected behavior

I should see `N` signs in sign column when `signcolumn=yes:N`. 

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1902+g184f84234

### Vim (not Nvim) behaves the same?

This feature does not exist in vim.

### Operating system/version

macOS Sonoma 14.2

### Terminal name/version

wezterm 20230712-072601-f4abf8fd

### $TERM environment variable

xterm-256color

### Installation

Bob neovim version manager

### PR DESCRIPTION
**Pull request by luukvbaal**
Fix #26685

### CODE REVIEW

**Comment by luukvbaal** [comment]
Not sure why this passes CI currently, there are plenty of tests that should fail. I'll try to figure out why they don't anymore and add one that does.

File: test/functional/ui/sign_spec.lua
Comment (lewis6991) [inline test/functional/ui/sign_spec.lua:1]: Changes like this should generally happen in separate PR's or commits so it makes reviews easier.

**Review by lewis6991 — APPROVED** [review]

File: test/functional/ui/sign_spec.lua
Comment (luukvbaal) [reply test/functional/ui/sign_spec.lua:1]: These changes caused a test failure without the code change in this PR, that's why I added them here. Not all of them were necessary though I suppose.

Comment (lewis6991) [reply test/functional/ui/sign_spec.lua:1]: It's mostly the `command` -> `exec` change that made this more difficult to review.

Comment (luukvbaal) [reply test/functional/ui/sign_spec.lua:1]: Well like I said that change was needed to execute multiple `:sign place` commands without validating in between.

**Pull request merged**